### PR TITLE
add `@testing-library/user-event` dev dependency

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

* Add `@testing-library/user-event` to Calypso dev deps

#### Testing instructions

* Unit tests shall pass
* No dependency errors

Related to #63409

### CODE REVIEW

**Comment by flootr** [comment]
Locally I see a warning from yarn which says

> calypso@workspace:client doesn't provide @testing-library/dom (pe4e2c), requested by @testing-library/user-event

As far as I understand `@testing-library/dom` should be provided by the React framework wrapper of testing-library or at least point to the same version which our currently used version of `@testing-library/react` is depending on (I'm not exactly sure how to ensure this).

For v14 (we're adding v13 in this PR) the official [documentation](https://testing-library.com/docs/user-event/install/) says:

> If you use one of the framework wrappers, it is important that @testing-library/dom is resolved to the same installation required by the framework wrapper of your choice.
> Usually this means that if you use one of the framework wrappers, you should not add @testing-library/dom to your project dependencies.

**Comment by tyxla** [comment]
I've added `@testing-library/dom` as a dev dependency, and it seems to be doing the job and tests pass. Perhaps we could leave it like that until we migrate to React 18?

**Comment by tyxla** [comment]
E2e tests seem to be failing consistently. I've rebased to see if that helps.

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Thanks for this, @flootr! I think it would be great if we can update the rest of the `fireEvent` calls.

File: client/package.json [hunk 1]
```diff
@@ -209,9 +209,11 @@
 	"devDependencies": {
 		"@automattic/calypso-eslint-overrides": "workspace:^",
 		"@sentry/webpack-plugin": "^1.18.8",
+		"@testing-library/dom": "^8.13.0",
 		"@testing-library/jest-dom": "^5.16.2",
 		"@testing-library/react": "^12.1.3",
 		"@testing-library/react-hooks": "7.0.2",
+		"@testing-library/user-event": "13.5.0",
```
Comment (ciampo) [inline client/package.json]: Just flagging that `user-event@13` is no longer maintained — current stable version is `v14` (which is also the version that docs refer to)

You may want to update the version before merging this PR?

Comment (tyxla) [reply client/package.json]: @ciampo in my chats with @flootr I had the understanding that v14 requires React 18, which is why we're using v13.

Comment (ciampo) [reply client/package.json]: Interesting point — [it seems so](https://github.com/testing-library/user-event/blob/main/package.json#L60-L61). 

Although Gutenberg is using [version 14](https://github.com/WordPress/gutenberg/blob/trunk/package.json#L115) while still using [react 17](https://github.com/WordPress/gutenberg/blob/trunk/package.json#L208-L209). Not sure what's going on 😅

**Comment by noahtallen** [comment]
> As far as I understand @testing-library/dom should be provided by the React framework wrapper of testing-library or at least point to the same version which our currently used version of @testing-library/react is depending on (I'm not exactly sure how to ensure this).

The way yarn works with peer dependencies, if `@testing-library/user-event` has dom as a peer dep, then dom must be added to our dependencies. If nothing else in the repo had an explicit dep on dom, then user-event wouldn't work at all.

I think we partially ensure this with `yarn dedupe`, so as long as the packages are relying on something within a similar semver range, dedupe will make sure only one dependency is installed.

It sounds like the root problem is that testing library is kind of inconsistent with when it uses peer deps vs not. :/

File: client/package.json (hunk 1 above)
Comment (flootr) [reply client/package.json]: @tyxla @ciampo I pushed an attempt to using v14 via [fd13abf](https://github.com/Automattic/wp-calypso/pull/63757/commits/fd13abfbd5bd9b03ff95623c114e63149bdc431e). Just wanted to check whether that'd cause any problems.

Comment (tyxla) [reply client/package.json]: Everything appears to be in order after the update! The e2e test failures seem unrelated.

**Pull request merged**